compose/ui/ui-text/src/uikitMain/kotlin/androidx/compose/ui/text/input/PlatformImeOptions.uikit.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.text.input
+
+import androidx.compose.runtime.Immutable
+import platform.UIKit.UIKeyboardAppearance
+import platform.UIKit.UIKeyboardAppearanceDefault
+import platform.UIKit.UIKeyboardType
+import platform.UIKit.UIKeyboardTypeDefault
+import platform.UIKit.UIReturnKeyType
+import platform.UIKit.UITextAutocapitalizationType
+import platform.UIKit.UITextAutocorrectionType
+import platform.UIKit.UITextContentType
+import platform.UIKit.UITextSpellCheckingType
+
+/**
+ * Used to configure iOS platform IME options.
+ *
+ * This class allows specifying UIKit-specific input method editor options. Note that any values
+ * explicitly specified here will override the corresponding settings from the enclosing [ImeOptions].
+ */
+@Immutable actual class PlatformImeOptions private constructor(
```

Comment:
Introducing such `expect`/`actual` public API will complicate moving iOS part into a separate library, so I'd avoid such things. It's bad to have such public `PlatformImeOptions` class in the first place, but we need to deal with it now

Alternative intermediate plan: expose to public API only things that can be moved outside.
- Free function instead of constructor
- Extensions instead of properties

Might be outside this PR: reshake internal structure and keep actual implementation only in skiko. Use separate interface as implementation details

Reply:
I have updated the API to make it less coupled to `PlatformImeOptions`. This commit https://github.com/JetBrains/compose-multiplatform-core/pull/2108/commits/cecf88bd5ee8817daefe71ede782bb0deb61c427 adds the actual implementations for the platform specific source-sets. I did this because to make `PlatformImeOptions` subclassable we need to make it `open`. We could revert this change and have only one `actual` implementation of `PlatformImeOptions` in `skikoMain` (as it was before) but then we would have to make that `open` and I don't know if we want that. Can you please check the current state and let me know if this better suits our needs @MatkovIvan ?

Reply:
> We could revert this change and have only one actual implementation of PlatformImeOptions in skikoMain (as it was before) but then we would have to make that open and I don't know if we want that.

Yes, let's do that. It's good that we can mark only one `actual` as `open`